enterprise/server/remote_execution/commandutil/commandutil.go

```diff
@@ -38,13 +38,62 @@ var (
 	ErrSIGKILL = status.UnavailableErrorf("command was terminated by SIGKILL, likely due to executor shutdown or OOM")
 
 	DebugStreamCommandOutputs = flag.Bool("debug_stream_command_outputs", false, "If true, stream command outputs to the terminal. Intended for debugging purposes only and should not be used in production.")
+	StdOutErrMaxSize          = flag.Uint64("executor.stdouterr_max_size_bytes", 0, "The maximum size of stdout/stderr for each action, in bytes. If the size of stdout/stderr exceeds this limit, the command will fail with a RESOURCE_EXHAUSTED error. If set to 0, no limit is enforced.")
 )
 
 var (
 	// Regexp matching a string consisting solely of digits (0-9).
 	allDigits = regexp.MustCompile(`^\d+$`)
 )
 
+func LimitStdOutErrWriter(w io.Writer) io.Writer {
+	if *StdOutErrMaxSize == 0 {
+		return w
+	}
+	return &limitWriter{w: w, limit: *StdOutErrMaxSize}
+}
+
+// limitWriter limits the number of bytes written to it.
+type limitWriter struct {
+	w     io.Writer
+	limit uint64
+
+	n uint64
+}
+
+func (lw *limitWriter) Write(p []byte) (int, error) {
```

Comment:
I think this can be simplified a bit

```

func (lw *limitWriter) Write(p []byte) (n int, err error) {
	capacity := min(lw.limit-lw.n, len(p))
	if capacity > 0 {
		n, err = lw.w.Write(p[:capacity])
		lw.n += n
	}
	

	// If we've reached the limit, return an error,
	// but don't mask the Write error if one occurred.
	if err == nil && capacity < len(p) {
		return n, status.ResourceExhaustedError("stdout/stderr size exceeded limit (%d bytes)", lw.limit)
	}

	return n, err
}
```

do tests pass with this implementation?